environments/builder/builder.go

```diff
@@ -0,0 +1,154 @@
+package builder
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/mholt/archiver"
+
+	"github.com/fission/fission"
+)
+
+const (
+	pathPrefix      = "/tmp/fission"
+	srcPkgPath      = pathPrefix + "/srcPkg"
+	deployPkgPath   = pathPrefix + "/deployPkg"
+	archiveFilename = "archive.zip"
+
+	// supported environment variables
+	envSrcPkg    = "SRC_PKG"
+	envDeployPkg = "DEPLOY_PKG"
+)
+
+type (
+	Builder struct {
+		sharedVolumePath string
+	}
+)
+
+func MakeBuilder(sharedVolumePath string) *Builder {
+	return &Builder{
+		sharedVolumePath: sharedVolumePath,
+	}
+}
+
+func (builder *Builder) Handler(w http.ResponseWriter, r *http.Request) {
```

Comment:
So it seems this builder is designed to do only one build at a time, but doesn't actually enforce that. If multiple concurrent builds are going to break, a sanity check in this handler would be helpful (you can use a global `sync/atomic` variable to track if a build is in progress).

Reply:
It's more reasonable to accept multiple builds, I will try to adopt that. And there is one question, does the builder pod (fetcher + builder) need to keep alive till the environment is removed?  
In my original design, builder pod (fetcher + builder) is ephemeral. Every time a build request comes to builder manager, it creates a pod to do the build, and remove it when it's in the idle state. Or we can treat the builder pod like a special environment that only for the build. So that builder manager can fire an HTTP request to trigger the build request.

@soamvasani what do you think about it?

Reply:
> accept multiple builds

sounds great!

> In my original design, builder pod (fetcher + builder) is ephemeral. Every time a build request comes to builder manager, it creates a pod to do the build,

Hmm. This question is more about the build manager, right? Does it affect this PR?